drivers/d3d12/rendering_device_driver_d3d12.cpp

```diff
@@ -4317,11 +4443,53 @@ void RenderingDeviceDriverD3D12::command_resolve_texture(CommandBufferID p_cmd_b
 }
 
 void RenderingDeviceDriverD3D12::command_clear_color_texture(CommandBufferID p_cmd_buffer, TextureID p_texture, TextureLayout p_texture_layout, const Color &p_color, const TextureSubresourceRange &p_subresources) {
-	const CommandBufferInfo *cmd_buf_info = (const CommandBufferInfo *)p_cmd_buffer.id;
 	TextureInfo *tex_info = (TextureInfo *)p_texture.id;
 
-	ID3D12Resource *res_to_clear = tex_info->main_texture ? tex_info->main_texture : tex_info->resource;
-	if ((tex_info->desc.Flags & D3D12_RESOURCE_FLAG_ALLOW_RENDER_TARGET)) {
+	_command_clear_color_texture(p_cmd_buffer, tex_info, p_color, p_subresources, true);
+}
+
+void RenderingDeviceDriverD3D12::_command_clear_color_texture(CommandBufferID p_cmd_buffer, TextureInfo *p_tex_info, const Color &p_color, const TextureSubresourceRange &p_subresources, bool p_use_transition_batching) {
+	const CommandBufferInfo *cmd_buf_info = (const CommandBufferInfo *)p_cmd_buffer.id;
+	ID3D12Resource *res_to_clear = p_tex_info->main_texture ? p_tex_info->main_texture : p_tex_info->resource;
+
+	auto _transition_subresources = [&](D3D12_RESOURCE_STATES p_new_state) {
+		D3D12_RESOURCE_BARRIER *non_batched_barriers = ALLOCA_ARRAY(D3D12_RESOURCE_BARRIER, p_subresources.layer_count * p_subresources.mipmap_count);
```

Comment:
It seems like these barriers are allocated regardless of whether you use transition batching or not. Shouldn't they only be allocated when it's false?

Reply:
In the next push I'm only allocating if it's `false`. Regarding the potentially huge count of barriers, I'd wait until it's known to be a real-world issue. The `alloca` is done with the worst case in mind, but it's very likely that many can be merged. As far as I know, `alloca` won't cause a crash by itself, only if addresses beyond the stack turned out to be actually used.